public/app/features/apiserver/client.ts

```diff
@@ -90,3 +97,13 @@ export class ScopedResourceClient<T = object, K = string> implements ResourceCli
       .join(',');
   }
 }
+
+// add the origin annotations so we know what was set from the UI
+function setOriginAsUI(meta: Partial<ObjectMeta>) {
+  if (!meta.annotations) {
+    meta.annotations = {};
+  }
+  meta.annotations.AnnoKeyOriginName = 'UI';
```

Comment:
Out of curiosity, where are those annotations sent to in the API server?

Reply:
[annotations](https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/) are a standard part of the k8s object's metadata, so they are passed on to storage automatically by the apiserver when creating or updating an object

Reply:
thanks @toddtreece !